### PR TITLE
Expose context and x_broker_api_originating_identity to brokerpaks

### PR DESCRIPTION
**Pull request by pivotal-marcela-campo**
[#177115261](https://www.pivotaltracker.com/story/show/177115261)

Implements https://github.com/cloudfoundry-incubator/cloud-service-broker/issues/138

### CODE REVIEW

**Review by gberche-orange — APPROVED** [review]
Awesome, thanks @pivotal-marcela-campo !

File: docs/brokerpak-specification.md [hunk 1]
```diff
@@ -462,7 +462,7 @@ Note that the order the variables are combined in code is slightly different.
 Moving default variables to be loaded third allow their computed values to make more sense.
 This is because they can resolve variables to the user's values first.
 
-#### Provision
```
Comment (gberche-orange) [inline docs/brokerpak-specification.md:465]: Should this also include "Update" (a service instance) ?

File: pkg/broker/service_definition.go [hunk 2]
```diff
@@ -402,14 +402,20 @@ func (svc *ServiceDefinition) variables( constants map[string]interface{},
 	return buildAndValidate(builder, svc.ProvisionInputVariables)
 }
 
-func (svc *ServiceDefinition) ProvisionVariables(instanceId string, details brokerapi.ProvisionDetails, plan ServicePlan) (*varcontext.VarContext, error) {
+func (svc *ServiceDefinition) ProvisionVariables(instanceId string, details brokerapi.ProvisionDetails, plan ServicePlan, originatingIdentity map[string]interface{}) (*varcontext.VarContext, error) {
```
Comment (gberche-orange) [inline pkg/broker/service_definition.go:405]: Should the `UpdateVariables` function also be updated to include the new fields ?

**Review by erniebilling — APPROVED** [review]

**Comment by erniebilling** [comment]
Fixes #138

**Pull request merged**